src/Features/Core/Portable/UseNullPropagation/AbstractUseNullPropagationDiagnosticAnalyzer.cs

```diff
@@ -110,10 +115,9 @@ private void AnalyzeSyntax(SyntaxNodeAnalysisContext context, INamedTypeSymbol e
                     syntaxFacts.GetOperandOfPrefixUnaryExpression(conditionNode));
             }
 
-            var isEqualityLikeCondition = TryAnalyzeCondition(
-                context, syntaxFacts, referenceEqualsMethodOpt, conditionNode,
-                out var conditionPartToCheck, out var isEquals);
-            if (!isEqualityLikeCondition)
+            if (!TryAnalyzeCondition(
+                    context, syntaxFacts, referenceEqualsMethodOpt, conditionNode,
+                    out var conditionPartToCheck, out var isEquals))
```

Comment:
needed for NRT to work.  if not directly in the 'if' check, the flow analysis doesn't know what is assigned or not later on.